Orange/widgets/unsupervised/owmds.py

```diff
@@ -748,150 +590,16 @@ def _update_plot(self):
-                brush_data[~self._subset_mask] = QBrush(Qt.NoBrush)
-
-            self._pen_data = pen_data
-            self._brush_data = brush_data
-
-        if self._shape_data is None:
-            shape_index = self.cb_shape_value.currentIndex()
-            if have_data and shape_index > 0:
-                Symbols = ScatterPlotItem.Symbols
-                symbols = numpy.array(list(Symbols.keys()))
-
-                shape_var = self.shapevar_model[shape_index]
-                data = column(self.data, shape_var, dtype=numpy.float)
-                data = data % (len(Symbols) - 1)
-                data[numpy.isnan(data)] = len(Symbols) - 1
-                shape_data = symbols[data.astype(int)]
-            else:
-                shape_data = "o"
-            self._shape_data = shape_data
-
-        if self._size_data is None:
-            MinPointSize = 3
-            point_size = self.symbol_size + MinPointSize
-            size_index = self.cb_size_value.currentIndex()
-            if have_data and size_index == 1:
-                # size by stress
-                size_data = stress(self.embedding, self._effective_matrix)
-                size_data = scale(size_data)
-                size_data = MinPointSize + size_data * point_size
-            elif have_data and size_index > 0:
-                size_var = self.sizevar_model[size_index]
-                size_data = column(self.data, size_var, dtype=float)
-                size_data = scale(size_data)
-                size_data = MinPointSize + size_data * point_size
-                size_data[numpy.isnan(size_data)] = 1  # maybe 0?
-            else:
-                size_data = point_size
-            self._size_data = size_data
-
-        if self._label_data is None:
-            label_index = self.cb_label_value.currentIndex()
-            if have_data and label_index > 0:
-                label_var = self.labelvar_model[label_index]
-                label_data = column(self.data, label_var)
-                label_data = [label_var.str_val(val) for val in label_data]
-                label_items = [pg.TextItem(text, anchor=(0.5, 0), color=0.0)
-                               for text in label_data]
-            elif self.matrix is not None and label_index:
-                attr = attributes(self.matrix)
-                label_items = [pg.TextItem(str(text), anchor=(0.5, 0),
-                                           color=0.0)
-                               for text in attr]
-            else:
-                label_items = None
-            self._label_data = label_items
-
         emb_x, emb_y = self.embedding[:, 0], self.embedding[:, 1]
-        if self.jitter > 0:
-            _, jitter_factor = self.JitterAmount[self.jitter]
-            emb_x = jitter(emb_x, jitter_factor, rstate=42)
-            emb_y = jitter(emb_y, jitter_factor, rstate=667)
 
-        if self.connected_pairs and self.__draw_similar_pairs:
+        def connected_pairs():
```

Comment:
Rename to `connect_pairs`. Perhaps move the initial `if` out of the function, so that `if` controls whether the function `connect_pairs` is called or not.